analyzers/src/ITs.JsonParser/IssueParser.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.IO;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using ITs.JsonParser.Json;
+using ITs.JsonParser.Reports;
+
+namespace ITs.JsonParser;
+
+/// <summary>
+/// Reads SARIF reports from "output/" and writes custom reports to "actual/".
+/// </summary>
+public static class IssueParser
+{
+    private static readonly JsonSerializerOptions SerializerOptions = new() { PropertyNameCaseInsensitive = true, WriteIndented = true };
+
+    public static void Execute(string inputRootPath, string outputRootPath)
+    {
+        Console.WriteLine($"Searching for SARIF reports in {inputRootPath}...");
+        var sarifReports = Read(inputRootPath);
+        ConsoleHelper.WriteLineColor("Successfully parsed all SARIF reports!", ConsoleColor.Green);
+
+        Console.WriteLine("Mapping SARIF reports to Rule reports..");
+        var customReports = Map(sarifReports);
+        ConsoleHelper.WriteLineColor($"Successfully mapped all SARIF reports to {customReports.Length} Rule reports!", ConsoleColor.Green);
+
+        Console.WriteLine($"Writing results to {outputRootPath}");
+        Write(outputRootPath, customReports);
+        ConsoleHelper.WriteLineColor("Successfully wrote all Rule reports.!", ConsoleColor.Green);
+    }
+
+    private static List<InputReport> Read(string rootPath)
```

Comment:
My original implementation was yield-all-the-way, but I opted for eager-execution to lazy, because of the logging.
If I don't evaluate the enumeration, the logging will be deferred and out of place.